smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/MiddlewareExecutionGenerator.kt

```diff
@@ -0,0 +1,151 @@
+package software.amazon.smithy.swift.codegen.integration
+
+import software.amazon.smithy.codegen.core.CodegenException
+import software.amazon.smithy.model.Model
+import software.amazon.smithy.model.knowledge.HttpBinding
+import software.amazon.smithy.model.knowledge.HttpBindingIndex
+import software.amazon.smithy.model.knowledge.OperationIndex
+import software.amazon.smithy.model.shapes.OperationShape
+import software.amazon.smithy.model.traits.EndpointTrait
+import software.amazon.smithy.model.traits.EnumTrait
+import software.amazon.smithy.model.traits.HttpChecksumRequiredTrait
+import software.amazon.smithy.model.traits.HttpTrait
+import software.amazon.smithy.model.traits.IdempotencyTokenTrait
+import software.amazon.smithy.model.traits.TimestampFormatTrait
+import software.amazon.smithy.swift.codegen.ClientRuntimeTypes
+import software.amazon.smithy.swift.codegen.IdempotencyTokenMiddlewareGenerator
+import software.amazon.smithy.swift.codegen.ServiceGenerator
+import software.amazon.smithy.swift.codegen.SwiftWriter
+import software.amazon.smithy.swift.codegen.integration.middlewares.ContentMD5Middleware
+import software.amazon.smithy.swift.codegen.model.camelCaseName
+import software.amazon.smithy.swift.codegen.model.capitalizedName
+import software.amazon.smithy.swift.codegen.model.hasTrait
+import software.amazon.smithy.swift.codegen.model.isBoxed
+import software.amazon.smithy.swift.codegen.model.toMemberNames
+import software.amazon.smithy.swift.codegen.swiftFunctionParameterIndent
+
+class MiddlewareExecutionGenerator(
+    private val ctx: ProtocolGenerator.GenerationContext,
+    private val writer: SwiftWriter,
+    private val httpBindingResolver: HttpBindingResolver,
+    private val defaultContentType: String,
+    private val httpProtocolCustomizable: HttpProtocolCustomizable,
+    private val operationStackName: String
+) {
+    private val model: Model = ctx.model
+    private val symbolProvider = ctx.symbolProvider
+
+    fun render(opIndex: OperationIndex, op: OperationShape, onError: (SwiftWriter, String) -> Unit) {
+        val httpTrait = httpBindingResolver.httpTrait(op)
+        val requestBindings = httpBindingResolver.requestBindings(op)
+        val pathBindings = requestBindings.filter { it.location == HttpBinding.Location.LABEL }
+        renderUriPath(httpTrait, pathBindings, writer, onError)
+
+        val operationErrorName = "${op.capitalizedName()}OutputError"
+        val inputShapeName = ServiceGenerator.getOperationInputShapeName(symbolProvider, opIndex, op)
+        val outputShapeName = ServiceGenerator.getOperationOutputShapeName(symbolProvider, opIndex, op)
+        writer.write("let context = \$N()", ClientRuntimeTypes.Http.HttpContextBuilder)
+        writer.swiftFunctionParameterIndent {
+            renderContextAttributes(op)
+        }
+        writer.write("var $operationStackName = OperationStack<$inputShapeName, $outputShapeName, $operationErrorName>(id: \"${op.camelCaseName()}\")")
```

Comment:
this looks like a spot I forgot to make this a runtime type -> OperationStack

Reply:
ah! on it.